pkg/router/template/router.go

```diff
@@ -203,6 +204,19 @@ func matchPattern(pattern, s string) bool {
 	return false
 }
 
+// Generate a regular expression to match wildcard hosts (and paths if any)
+// for a [sub]domain.
+func genDomainWildcardRegexp(hostname, path string, exactPath bool) string {
+	route := &routeapi.Route{Spec: routeapi.RouteSpec{Host: hostname}}
+	host, _ := validation.NormalizeWildcardRoute(route)
+	expr := regexp.QuoteMeta(fmt.Sprintf(".%s%s", host, path))
```

Comment:
The quoting should but its RE2 vs PCRE. That said though, the only meta character we are using here is `*` and `.` - which is ok.